### PR TITLE
Document regex capture groups in rules blocks

### PR DESCRIPTION
**Pull request by Crell**
Someone from engineering please confirm that this is what we did. :smile:

### CODE REVIEW

File: src/configuration/app-containers.md [hunk 1]
```diff
@@ -316,20 +316,45 @@ web:
             passthru: "/index.php"
             index:
                 - index.php
+            # No caching, unless the PHP script sends a cache header.
```
Comment (pjcdawkins) [inline src/configuration/app-containers.md]: I thought `expires` only applies to static files, and doesn't affect scripts at all?

Comment (Crell) [inline src/configuration/app-containers.md]: Isn't that implied in that comment? Could we clarify it further?

Comment (pjcdawkins) [inline src/configuration/app-containers.md]: > No caching (for static files).

**Comment by pjcdawkins** [comment]
I corrected the regex, and because we have examples of YAML regular expressions on that page, I changed all the YAML in the file to single quotes instead of double quotes. Someone else will need to review this now.

File: src/configuration/app-containers.md [hunk 2]
```diff
@@ -462,7 +483,7 @@ Your `.platform.app.yaml` file should include:
 ```yaml
 hooks:
   deploy: |
-    if [ $PLATFORM_BRANCH = "master" ]; then
+    if [ "$PLATFORM_BRANCH" = master ]; then
```
Comment (Crell) [inline src/configuration/app-containers.md]: The "master" doesn't need to be quoted?  Are you sure about that?

Comment (pjcdawkins) [inline src/configuration/app-containers.md]: Yes (we know exactly what's in the string, so the quotes there are superfluous, whereas we know less about what would be in the PLATFORM_BRANCH variable)

File: src/configuration/app-containers.md [hunk 3]
```diff
@@ -306,30 +301,56 @@ It has a few subkeys listed below:
...
             expires: 300
             passthru: true
+            allow: false
+            rules:
+                # Only allow static image files from the images directory.
+                '\.(jpe?g|png|gif|svgz?|ico|bmp)$':
+                    allow: true
+```
+
+### [Example] Advanced rewrite rules
```
Comment (mrkschan) [inline src/configuration/app-containers.md]: @sruggier Is this block superseding https://docs.platform.sh/configuration/app-containers.html#url-rewriting?

**Pull request merged**